CAIRN Info.js

```diff
@@ -67,44 +67,63 @@ function doWeb(doc,url)
 			for (var j in items) {
 				urls.push(j);
 			}
-			ZU.processDocuments(urls, function (myDoc) { 
-				doWeb(myDoc, myDoc.location.href) });
-
+			ZU.processDocuments(urls, scrape);
 		});
 	} else {
-		// We call the Embedded Metadata translator to do the actual work
-		var translator = Zotero.loadTranslator("import");
-		translator.setTranslator("951c027d-74ac-47d4-a107-9c3069ab7b48");
-		translator.setHandler("itemDone", function(obj, item) {
-				item.tags= [];
-				var keywords = ZU.xpathText(doc, '//meta[@name="citation_keyword"]/@content');
-				if (keywords) keywords = keywords.split(/\s*[,;]\s*/);
-				for (i in keywords){
-					if (keywords[i].search(/[^\s]/) != -1){
-						tags.push(keywords[i])
-					}
-				}
-
-				// The default value for PDF download is on an HTML page that calls the actual download
-				// we need to add this attachment after the import translator has been running
-				// We remove the attached PDF because they won't be working
-				for(var i=item.attachments.length - 1; i>=0; i--) {
-					if(item.attachments[i].mimeType == 'application/pdf') {
-						item.attachments.splice(i, 1);
-					}
-				}
-				
-				var pdfUrl = ZU.xpathText(doc, '//meta[@name="citation_pdf_url"]/@content');
-				pdfUrlDownload = pdfUrl + "&download=1";
-				item.attachments.push({title : "Full Text PDF", mimeType: "application/pdf", url: pdfUrlDownload});
-				
-				item.complete();
-				});
-		translator.getTranslatorObject(function (obj) {
-				obj.doWeb(doc, url);
-				});
+		scrape(doc);
 	}
 }
+
+
+function scrape(doc)
+{
+	// We call the Embedded Metadata translator to do the actual work
+	var translator = Zotero.loadTranslator("web");
+	translator.setTranslator("951c027d-74ac-47d4-a107-9c3069ab7b48");
+	translator.setDocument(doc);
+	translator.setHandler("itemDone", function(obj, item) {
+		// Cairn.info uses non-standard keywords:
+		// we import them here, as the Embedded Metadata translator
+		// cannot catch them.
+		item.tags = [];
+		var keywords = ZU.xpathText(doc, '//meta[@name="article-mot_cle"]/@content');
+		if (keywords) keywords = keywords.split(/\s*[,;]\s*/);
+		for (i in keywords) {
+			if (keywords[i].search(/[^\s]/) != -1) {
```

Comment:
`\S` == `[^\s]`, but `if (keywords[i].trim())` should do the same and, I think, is a bit clearer.